### ISSUE TITLE
bulkWrite throws an error when set `timestamps: false` in schema

### ISSUE THREAD

**Issue by SoftwareSing** [Automattic/mongoose#10048]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**

When set `timestamps: false` in schema, `bulkWrite` with `insertOne` or `replaceOne` will throw an error.
But it works fine when set `timestamps: null` or just don't set timestamps.

**If the current behavior is a bug, please provide the steps to reproduce.**

```javascript
const mongoose = require('mongoose')

module.exports = async function bulkInsertOne () {
  const schema = new mongoose.Schema({ name: String }, { timestamps: false })
  const Model = mongoose.model('MyModel', schema)
  await Model.bulkWrite([{
    insertOne: {
      document: { name: 'test' }
    }
  }])
}
```

error:

```
(node:24085) UnhandledPromiseRejectionWarning: TypeError: doc.initializeTimestamps is not a function
    at /home/miku/_____/code/test-for-mongoose-timestamps/node_modules/mongoose/lib/helpers/model/castBulkWrite.js:24:13
    at /home/miku/_____/code/test-for-mongoose-timestamps/node_modules/mongoose/lib/model.js:3500:35
    at each (/home/miku/_____/code/test-for-mongoose-timestamps/node_modules/mongoose/lib/helpers/each.js:11:5)
    at /home/miku/_____/code/test-for-mongoose-timestamps/node_modules/mongoose/lib/model.js:3500:5
    at /home/miku/_____/code/test-for-mongoose-timestamps/node_modules/mongoose/lib/helpers/promiseOrCallback.js:31:5
    at new Promise (<anonymous>)
    at promiseOrCallback (/home/miku/_____/code/test-for-mongoose-timestamps/node_modules/mongoose/lib/helpers/promiseOrCallback.js:30:10)
    at Mongoose._promiseOrCallback (/home/miku/_____/code/test-for-mongoose-timestamps/node_modules/mongoose/lib/index.js:1149:10)
    at Function.Model.bulkWrite (/home/miku/_____/code/test-for-mongoose-timestamps/node_modules/mongoose/lib/model.js:3498:23)
    at bulkInsertOne (/home/miku/_____/code/test-for-mongoose-timestamps/src/bulkInsertOne.js:6:15)
    at main (/home/miku/_____/code/test-for-mongoose-timestamps/src/main.js:7:9)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**What is the expected behavior?**

It should not throw any error.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**

- Node.js version: 14.16.0
- Mongoose version: 5.12.1
- MongoDB version: 4.2.13

**Comment by vkarpov15** [comment]
@SoftwareSing the below script executes without error. Please modify the below script to demonstrate your issue:

```javascript
const mongoose = require('mongoose');

mongoose.set('debug', true);

run().catch(err => console.log(err));

async function run() {
  await mongoose.connect('mongodb://localhost:27017/test', {
    useNewUrlParser: true,
    useUnifiedTopology: true
  });

  await mongoose.connection.dropDatabase();

  const schema = new mongoose.Schema({ name: String }, { timestamps: false });
  const Model = mongoose.model('MyModel', schema)
  await Model.bulkWrite([{
    insertOne: {
      document: { name: 'test' }
    }
  }]);

  console.log('Done');
}
```

**Comment by SoftwareSing** [comment]
@vkarpov15 well...... it's already fixed by #10050 , I don't know why this issue is reopened.
